drv/sidecar-mainboard-controller/src/ignition.rs

```diff
@@ -0,0 +1,143 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use crate::{Addr, MainboardController};
+use drv_fpga_api::{FpgaError, FpgaUserDesign, WriteOp};
+use drv_ignition_api::*;
+use zerocopy::{AsBytes, FromBytes};
+
+pub struct IgnitionController {
+    fpga: FpgaUserDesign,
+}
+
+impl IgnitionController {
+    pub fn new(task_port: userlib::TaskId) -> Self {
+        Self {
+            fpga: FpgaUserDesign::new(
+                task_port,
+                MainboardController::DEVICE_INDEX,
+            ),
+        }
+    }
+
+    #[inline]
+    fn port_addr(&self, port: u8, offset: IgnitionAddr) -> u16 {
+        u16::from(Addr::IGNITION_CONTROLLERS_COUNT)
+            + 0x100
+            + (0x100 * port as u16)
+            + u16::from(offset)
+    }
+
+    fn read_port_register<T>(
```

Comment:
The methods these call on `FpgaUserDesign` may also need `inline` to see a difference (they're small generic wrappers around non-generic implementations).

Reply:
The compiler may have already reached the conclusion that all this can be inlined since explicitly requesting it doesn't seem to change the size. But either way it seems like a reasonable thing to do.